app/routes/circulars/circulars.server.ts

```diff
@@ -52,6 +52,40 @@ function parseDate(date?: string) {
   return date ? new Date(date).getTime() : NaN
 }
 
+/** take input string and return start/end times based on string value */
+function fuzzyTimeRange(fuzzyTime?: string) {
+  const now = Date.now()
+  if (fuzzyTime === 'now') return now
+  if (fuzzyTime === 'hour') return now - 3600000
+  if (fuzzyTime === 'today') return new Date().setHours(0, 0, 0, 0)
+  if (fuzzyTime === 'day') return now - 86400000
+  if (fuzzyTime === 'week') return now - 86400000 * 7
+  if (fuzzyTime === 'month') return now - 86400000 * 30
+  if (fuzzyTime === 'year') return now - 86400000 * 365
+  if (fuzzyTime === 'mtd')
+    return new Date(
+      new Date().getFullYear(),
+      new Date().getMonth(),
+      1
+    ).getTime()
+  if (fuzzyTime === 'ytd')
+    return new Date(new Date().getFullYear(), 0).getTime()
+  else return undefined // invalid fuzzyTime defaults to fuzzless time range
+}
+
+function getValidDates(startDate?: string, endDate?: string) {
+  // regex for YYYY-MM-DD and another for MM/DD/YYYY
+  const usDateRegex = /(\d{4})-(\d{2})-(\d{2})|(\d{2})\/(\d{2})\/(\d{4})/
+  const normalDateRegex = /(\d{4})-(\d{2})-(\d{2})/
+  const usDateMatch = startDate?.match(usDateRegex)
+  const normalDateMatch = startDate?.match(normalDateRegex)
+  if (usDateMatch || normalDateMatch) {
+    return [parseDate(startDate), parseDate(endDate) + 86400000]
+  } else {
+    return [fuzzyTimeRange(startDate), fuzzyTimeRange(endDate)]
+  }
```

Comment:
There's no need for these regular expressions. You can first call `fuzzyTimeRange`, and parse the dates if its return value is undefined.

Reply:
ah, good point. Addressed in [8f2e6ef](https://github.com/nasa-gcn/gcn.nasa.gov/pull/1172/commits/8f2e6efbdd8af3b4492be45cdc7de9b38276185b)